src/main/logger.ts

```diff
@@ -1,21 +1,30 @@
 import path from 'path'
 import _ from 'lodash/fp'
+import ElectronLog from 'electron-log'
 import {config} from '../config/main'
 import {createLogger} from '../shared/utils'
-import {store} from './store'
+import {MainStore} from './store'
+import {DatadirChecked} from './compatibility-check'
 
-export const mainLog = createLogger('main', () =>
-  path.join(config.dataDir, 'logs', store.get('networkName'), 'main.log'),
-)
+export const createMainLog = (
+  _datadirChecked: DatadirChecked,
+  store: MainStore,
+): ElectronLog.ElectronLog => {
+  const mainLog = createLogger('main', () =>
+    path.join(config.dataDir, 'logs', store.get('networkName'), 'main.log'),
```

Comment:
1. since that's the only way store is used here - maybe just network name should be received?
2. Is the logger recreated each time mantis is restarted?
3. please note that if datadir param contains path to datadir too - there's no need to read it from config additionally (and again - config should come from arguments)

Reply:
Good points

Reply:
I reverted your previous suggestion regarding passing only `networkName` to `createMainLog` instead of the `store`:

The logger's `resolvePath` prop is a function. It's called every time when logs are saved to disk.
If we pass `store`, `resolvePath` can fetch the appropriate path automatically, so we don't need to recreate the logger when switching between networks.